backend/proxy/secure_socks_proxy.go

```diff
@@ -0,0 +1,212 @@
+package proxy
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"golang.org/x/net/proxy"
+)
+
+var (
+
+	// PluginSecureSocksProxyEnabled is a constant for the GF_SECURE_SOCKS_DATASOURCE_PROXY_SERVER_ENABLED
+	// environment variable used to specify if a secure socks proxy is allowed to be used for datasource connections.
+	PluginSecureSocksProxyEnabled = "GF_SECURE_SOCKS_DATASOURCE_PROXY_SERVER_ENABLED"
+	// PluginSecureSocksProxyClientCert is a constant for the GF_SECURE_SOCKS_DATASOURCE_PROXY_CLIENT_CERT
+	// environment variable used to specify the file location of the client cert for the secure socks proxy.
+	PluginSecureSocksProxyClientCert = "GF_SECURE_SOCKS_DATASOURCE_PROXY_CLIENT_CERT"
+	// PluginSecureSocksProxyClientKey is a constant for the GF_SECURE_SOCKS_DATASOURCE_PROXY_CLIENT_KEY
+	// environment variable used to specify the file location of the client key for the secure socks proxy.
+	PluginSecureSocksProxyClientKey = "GF_SECURE_SOCKS_DATASOURCE_PROXY_CLIENT_KEY"
+	// PluginSecureSocksProxyRootCACert is a constant for the GF_SECURE_SOCKS_DATASOURCE_PROXY_ROOT_CA_CERT
+	// environment variable used to specify the file location of the root ca for the secure socks proxy.
+	PluginSecureSocksProxyRootCACert = "GF_SECURE_SOCKS_DATASOURCE_PROXY_ROOT_CA_CERT"
+	// PluginSecureSocksProxyProxyAddress is a constant for the GF_SECURE_SOCKS_DATASOURCE_PROXY_PROXY_ADDRESS
+	// environment variable used to specify the secure socks proxy server address to proxy the connections to.
+	PluginSecureSocksProxyProxyAddress = "GF_SECURE_SOCKS_DATASOURCE_PROXY_PROXY_ADDRESS"
+	// PluginSecureSocksProxyServerName is a constant for the GF_SECURE_SOCKS_DATASOURCE_PROXY_SERVER_NAME
+	// environment variable used to specify the server name of the secure socks proxy.
+	PluginSecureSocksProxyServerName = "GF_SECURE_SOCKS_DATASOURCE_PROXY_SERVER_NAME"
+)
+
+// SecureSocksProxyConfig contains the information needed to allow datasource connections to be
+// proxied to a secure socks proxy
+type secureSocksProxyConfig struct {
+	clientCert   string
+	clientKey    string
+	rootCA       string
+	proxyAddress string
+	serverName   string
+}
+
+// SecureSocksProxyEnabled checks if the Grafana instance allows the secure socks proxy to be used
+// and the datasource options specify to use the proxy
+func SecureSocksProxyEnabled(opts *Options) bool {
+	if opts == nil {
+		return false
+	}
+
+	if !opts.EnabledOnDS {
+		return false
+	}
+
+	if value, ok := os.LookupEnv(PluginSecureSocksProxyEnabled); ok {
+		enabledOnInst, err := strconv.ParseBool(value)
+		if err != nil {
+			return false
+		}
+
+		return enabledOnInst
+	}
+
+	return false
+}
+
+// SecureSocksProxyEnabledOnDS checks the datasource json data for `enableSecureSocksProxy`
+// to determine if the secure socks proxy should be enabled on it
+func SecureSocksProxyEnabledOnDS(jsonData map[string]interface{}) bool {
```

Comment:
Is this still needed? I guess so, otherwise you would have to use the https://github.com/grafana/grafana-plugin-sdk-go/blob/603cb9577507bfd3a1d65a94f790b531e68cd8ee/backend/common.go#L100, right?

Reply:
Right - I was thinking this would be a helpful helper function for datasources